index.js

```diff
@@ -1,121 +1,74 @@
-'use strict';
 const request = require('request-promise-native');
 
 const role_to_use = process.env.ROLE;
 const couch_url = process.env.COUCH_URL;
-let compiledUrl = {};
 
-if(role_to_use === undefined){
+if(role_to_use === undefined) {
     console.log('ROLE is not set, please set it and try again');
     process.exit(1);
 }
-if(couch_url === undefined){
+if(couch_url === undefined) {
     console.log('COUCH_URL is not set, please set it and try again');
     process.exit(1);
 }
 
-const handleError = function(e){
-    if (e.statusCode === 401) {
-        console.log('Bad authentication for CouchDB. Check that COUCH_URL has correct usernames and passwords.');
-        process.exit(1);
-    } else {
-        console.log('Error connecting to CouchDB: ' + e.message);
-        process.exit(1);
-    }
-}
```

Comment:
It's cleaner to let errors bubble up to the error handling code than call a handling function from multiple places. At the very least, it makes it obvious where the code can exit!

Reply:
This is is one of my complaints about node - you get a wall of text (ok, not as bad as java, but still) and it's not clear what went wrong.  I liked to do as much hand holding as possible, especially when the code will be run interactively on the CLI.

Let's finish out this PR so it's all merged, but I'm curious what a node best practice is!

Reply:
> you get a wall of text

You mean the stacktrace in the console? The stacktrace is set when the `Error` is created, so it shouldn't matter whether you catch it right away (your option) or let it fall through to the ultimate `.catch` (my option).